TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/EditTaskScreen/EditTaskScreenViewController.m

```diff
@@ -0,0 +1,359 @@
+//
+//  EditTaskScreenViewController.m
+//  TestAssignmentDoItSoftware
+//
+//  Created by Vladimir Samoylenko on 2/15/19.
+//  Copyright © 2019 TheRedQueen. All rights reserved.
+//
+
+#import "EditTaskScreenViewController.h"
+
+@interface EditTaskScreenViewController ()
+
+@end
+
+@implementation EditTaskScreenViewController
+
+
+@synthesize lblDueTo;
+@synthesize btnSaveTask;
+@synthesize swNotification;
+@synthesize lblLineUnderNotification;
+@synthesize tvTaskTitle;
+@synthesize tvDescription;
+@synthesize btnDateDueForTask;
+@synthesize scPriority;
+
+
+#pragma mark - UI view methods
+
+
+- (void)viewDidLoad {
+
+    [super viewDidLoad];
+    
+    self.navigationController.navigationBar.topItem.title = @"Back";
+    
+    self.navigationItem.title = @"Edit task";
+    self.navigationController.navigationBar.translucent = NO;
+    self.navigationController.navigationBar.barTintColor = [UIColor colorWithRed:232.0/255.0 green:232.0/255.0 blue:232.0/255.0 alpha:1.0];
+    self.navigationController.navigationBar.tintColor = [UIColor colorWithRed:130.0/255.0 green:130.0/255.0 blue:130.0/255.0 alpha:1.0];
+
+    dateDue = [NSNumber numberWithInteger:0];
+    
+    tid = [NSNumber numberWithUnsignedInteger:0];
+    lid = [NSNumber numberWithUnsignedInteger:0];
+    priority = [NSNumber numberWithUnsignedInteger:1];
+    nid = [NSNumber numberWithUnsignedInteger:0];
+    status = [NSNumber numberWithUnsignedInteger:1];
+    title = @"";
+    details = @"";
+    prioritystr = @"";
+
+    tvTaskTitle.layer.borderColor = [UIColor blackColor].CGColor;
+    tvTaskTitle.layer.borderWidth = 1.0;
+    tvTaskTitle.layer.cornerRadius = 2.0;
+    
+    tvDescription.layer.borderColor = [UIColor blackColor].CGColor;
+    tvDescription.layer.borderWidth = 1.0;
+    tvDescription.layer.cornerRadius = 2.0;
+
+    shiftValue = [NSNumber numberWithInt:175];
+    
+}
+
+
+-(void)viewWillAppear:(BOOL)animated{
+    
+    [super viewWillAppear:animated];
+    
+    if ([dateDue unsignedIntegerValue] > 0){
+        
+        [self formatDueDate];
+        
+        return;
+    }
+    
+    NSArray *keys = [dictionaryTask allKeys];
+    
+    if ([keys count] == 0){
+        
+        TASingleton *singleton = [[TASingleton alloc] init];
+        
+        if ([lid unsignedIntegerValue] == 0){
+            lid = [singleton getNewId];
+        }
+        
+        priority = [NSNumber numberWithUnsignedInteger:0];
+        [self scPriorityChanged:scPriority];
+        
+        NSDate *from1970 = [[NSDate alloc] initWithTimeIntervalSinceNow:86400];
```

Comment:
The variable name is literally misleading as it doesn't contain a value from the epoch (instead it's based on the current timestamp)